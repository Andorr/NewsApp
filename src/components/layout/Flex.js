// @flow
import * as React from 'react';
import {withStyles} from '@material-ui/core/styles';
import classNames from 'classnames';

// Material UI Components

// Project Components

type P = {
    classes: Object,
    children: any,
    className: string,

    justify: ?string,
    align: ?string,
    dir: ?string,
}

const styles: Object = {
    root: {
        display: 'flex',
    }
}

const Flex: React.StatelessFunctionalComponent<P> = (props) => {
    const {classes} = props;
    const flexClass : Object = {
        ...classes,
        justifyContent: props.justify || 'flex-start',
        alignItems: props.align || 'center',
        flexDirection: props.dir || 'row',
    };

    return (
        <div className={classNames(classes.root, props.className)} style={flexClass}>
            {props.children}
        </div>
    )
}

export default withStyles(styles)(Flex);