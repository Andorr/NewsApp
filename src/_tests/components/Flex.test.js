import React from 'react';
import {shallow, mount} from 'enzyme';
import { createShallow , type ReactWrapper } from '@material-ui/core/test-utils';

import Flex from '../../components/layout/Flex';

describe('Flex component', () => {
    let shallow: ReactWrapper;
    
    beforeEach(() => {
        shallow = createShallow(); // Necassary for Material-UI's "withStyles()"
    });

    it('[Flex] Testing render', () => {
        const component: ReactWrapper = shallow(<Flex />);
        expect(component).toMatchSnapshot();
    });
});