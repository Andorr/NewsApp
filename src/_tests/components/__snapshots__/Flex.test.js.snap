// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex component [Flex] Testing render 1`] = `
<Flex
  classes={
    Object {
      "root": "Flex-root-1",
    }
  }
/>
`;
